### ISSUE TITLE
Support using functions to generate baseContents

### ISSUE THREAD

**Issue by raingerber** [vladshcherbin/rollup-plugin-generate-package-json#6]
Hello, and thanks for making this library! I forked it so I can use a function for generating `baseContents`, but if you're interested I'll open a PR to add this feature (and if not, thanks again 😀):

```js
generatePackageJson({
  baseContents: (pkg) => {
    // "pkg" is the contents of the inputFile
    return {
      name: pkg.name
    }
  }
})
```

**Comment by vladshcherbin** [comment]
@raingerber hey, there was a similar issue and suggestion here - https://github.com/vladshcherbin/rollup-plugin-generate-package-json/issues/1#issuecomment-443199660

The idea is this:

```js
import generatePackageJson from 'rollup-plugin-generate-package-json'
import { name } from './package.json'

export default {
  input: 'src/index.js',
  output: {
    file: 'dist/app.js',
    format: 'cjs'
  },
  plugins: [
    generatePackageJson({
      baseContents: {
        name
      }
    })
  ]
}
```

What do you think about it? If it's not good enough, I can definitely think about your proposal :)

**Comment by vladshcherbin** [comment]
Actually, nevermind, I'll think about the api for this functionality and add it in next version this week. Thanks for the suggestion :)

**Comment by raingerber** [comment]
Thanks, I like this approach because it uses the existing `inputFolder` option. If you're curious, this is the change I made on my fork https://github.com/raingerber/rollup-plugin-generate-package-json/commit/7377b87677f1d5158a6dcd77f876ef1a791340fb - I also thought about an array of keys like this, where the keys are used to filter the input `package.json`:

```js
{ 
  // not sure about the name "pickBy",
  // but it's just an example
  pickBy: ['name', 'description']
}
```

**Comment by vladshcherbin** [comment]
Released in 3.2.0

```js
generatePackageJson({
  baseContents: (pkg) => ({
    name: pkg.name,
    main: pkg.main.replace('src', 'dist')
    dependencies: {},
    private: true
  })
})
```

Sorry it took so long to add this feature 😞
